utils/s3/client.cc

```diff
@@ -250,26 +261,59 @@ storage_io_error map_s3_client_exception(std::exception_ptr ex) {
         auto e = std::current_exception();
         return {EIO, format("S3 error ({})", e)};
     }
+}
+
+future<> client::do_retryable_request(group_client& gc, http::request req, http::experimental::client::reply_handler handler) const {
+    uint32_t retries = 0;
+    std::exception_ptr e;
+    aws::aws_exception request_ex{aws::aws_error{aws::aws_error_type::OK, aws::retryable::yes}};
+    while (_retry_strategy->should_retry(request_ex.error(), retries)) {
```

Comment:
That's pretty ... horrible :( Prior to making even a single request the code calls retry strategy _and_ seastar::sleep with 0 timeout. While "fast path" should be -- call http::client::make_request() immediately and then sleep and go on another lap if error happens

Reply:
done

Reply:
Thanks, the new loop is

```c++
    do {
        try {
            e = {};
            co_return co_await gc.http.make_request(req, handler, std::nullopt);
        } catch (const aws::aws_exception& ex) {
            e = std::current_exception();
            request_ex = ex;
        }
        co_await seastar::sleep(_retry_strategy->delay_before_retry(request_ex.error(), retries));
        ++retries;
    } while (_retry_strategy->should_retry(request_ex.error(), retries));
```

and now if it's doing the final retry which fails, it will sleep before failing the whole thing :stuck_out_tongue: